bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/device/CommandHandler.java

```diff
@@ -0,0 +1,912 @@
+                Msg m = null;
+                if (cmd == OnOffType.ON) {
+                    level = getMaxLightLevel(conf, 0xff);
+                    direc = 0x11;
+                    logger.debug("{}: sent msg to switch {} to {}", nm(), dev.getAddress(),
+                            level == 0xff ? "on" : level);
+                } else if (cmd == OnOffType.OFF) {
+                    direc = 0x13;
+                    logger.debug("{}: sent msg to switch {} off", nm(), dev.getAddress());
+                }
+                if (ext == 1 || ext == 2) {
+                    byte[] data = new byte[] { (byte) getIntParameter("d1", 0), (byte) getIntParameter("d2", 0),
+                            (byte) getIntParameter("d3", 0) };
+                    m = dev.makeExtendedMessage((byte) 0x0f, (byte) direc, (byte) level, data);
+                    logger.debug("{}: was an extended message for device {}", nm(), dev.getAddress());
+                    if (ext == 1) {
+                        m.setCRC();
+                    } else if (ext == 2) {
+                        m.setCRC2();
+                    }
+                } else {
+                    m = dev.makeStandardMessage((byte) 0x0f, (byte) direc, (byte) level, getGroup(conf));
+                }
+                logger.debug("Sending message to {}", dev.getAddress());
+                dev.enqueueMessage(m, feature);
+                // expect to get a direct ack after this!
+            } catch (IOException e) {
+                logger.warn("{}: command send i/o error: ", nm(), e);
+            } catch (FieldException e) {
+                logger.warn("{}: command send message creation error ", nm(), e);
+            }
+        }
+    }
+
+    @NonNullByDefault
+    public static class FastOnOffCommandHandler extends CommandHandler {
+        FastOnOffCommandHandler(DeviceFeature f) {
+            super(f);
+        }
+
+        @Override
+        public void handleCommand(InsteonChannelConfiguration conf, Command cmd, InsteonDevice dev) {
+            try {
+                if (cmd == OnOffType.ON) {
+                    int level = getMaxLightLevel(conf, 0xff);
+                    Msg m = dev.makeStandardMessage((byte) 0x0f, (byte) 0x12, (byte) level, getGroup(conf));
+                    dev.enqueueMessage(m, feature);
+                    logger.debug("{}: sent fast on to switch {} level {}", nm(), dev.getAddress(),
+                            level == 0xff ? "on" : level);
+                } else if (cmd == OnOffType.OFF) {
+                    Msg m = dev.makeStandardMessage((byte) 0x0f, (byte) 0x14, (byte) 0x00, getGroup(conf));
+                    dev.enqueueMessage(m, feature);
+                    logger.debug("{}: sent fast off to switch {}", nm(), dev.getAddress());
+                }
+                // expect to get a direct ack after this!
+            } catch (IOException e) {
+                logger.warn("{}: command send i/o error: ", nm(), e);
```

Comment:
This is actually here because Msg.makeMessage() throws an IOException if it can't find the message. This should never happen, except during development. Can I leave as is?

Reply:
done, created custom exception to replace IOException and left as warn.